Entrega3

ElBrujas opened the pull request:
Modelo terminado (falta un poco de refactor)...

Jonathan-Vasquez committed:
reseteo total

AbrahamOsco committed:
Creo mi rama de trabajo abraham

AbrahamOsco committed:
elimino lo que hice

AbrahamOsco committed:
Probando rama de jhonathan

Jonathan-Vasquez committed:
probando 123

Jonathan-Vasquez committed:
borrando el ejemplo de abraham

AbrahamOsco committed:
Se agrego la interfaz Evaluable y sus clases herederas OpcionCorrecta,OpcionIncorrecta,OpcionPertenecia

AbrahamOsco committed:
Merge branch 'abraham' of https://github.com/AbrahamOsco/tp2Algo3 into abraham

marco-amm committed:
planteo inicial de Jugador y Partida

Commit:
firmas hechas

Commit:
firmas hechas

Jonathan-Vasquez committed:
se refactorizo el metodo evaluarRespuesta para no repetir codigo

Commit:
arreglado puntajesDeEstaRonda

Commit:
arregladas algunas cosas

Commit:
arregladas otras cosas

Commit:
modificaciones posibles en asignadorPuntos

marco-amm committed:
nada

Jonathan-Vasquez committed:
se refactorizo para ver que preguntas repiten el criterio de asignar puntos

Jonathan-Vasquez committed:
Merge branch 'Jonathan(Jet)' into Entrega2

marco-amm committed:
Merge branch 'marco' into Entrega2

Commit:
modificaciones segun nuevo diagrama de clases

Commit:
Merge branch 'Mateo' into Entrega2

AbrahamOsco committed:
Merge branch 'abraham' into Entrega2

AbrahamOsco committed:
Movi las clases al directorio Evaluable

Jonathan-Vasquez committed:
se borro la clase Menssage y su test

Jonathan-Vasquez committed:
Merge branch 'Entrega2' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega2

Jonathan-Vasquez committed:
se borro  una clase de abran

Jonathan-Vasquez committed:
cambios2

AbrahamOsco committed:
Agrego la interfaz ModificadorGlobal

Jonathan-Vasquez committed:
se dividio pregunta en PreguntaClasico,PregutnaPenalizable y PreguntaPuntajeParcial

Commit:
Se implemento por completo AsignadorPuntos según nuevo diagrama de clases

AbrahamOsco committed:
guardando cambios

Commit:
Merge branch 'Mateo' into Entrega2

AbrahamOsco committed:
Merge branch 'Entrega2' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega2

Jonathan-Vasquez committed:
se agrego modifcadores al jugador con HashMap

Commit:
Arreglados import en Ronda y AsignadorPuntos

Commit:
Merge branch 'Entrega2' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega2

AbrahamOsco committed:
Merge branch 'Entrega2' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega2

Commit:
Inicios de reTests entrega 0

Commit:
Merge branch 'Entrega2' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega2

Commit:
Arreglo de conflicto

Commit:
Solucionando conflicto

Commit:
Se añaden paquetes de test de entrega

AbrahamOsco committed:
agregando tests

Commit:
Añadimos Respuesta para hacer los test

Commit:
Añadido un método en Pregunta y corregida una letra

Commit:
Añadido un comentario

Commit:
Añadidos constructores de Opcion

Jonathan-Vasquez committed:
se agregaron los contructores de Preguntas

Jonathan-Vasquez committed:
Merge branch 'Entrega2' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega2

Commit:
Solucionados detalles en Ronda y AsignadorPuntos

Commit:
Corregidas dos lineas de Ronda

Commit:
Varias cosas para correr test. corre el primero correctamente.

Jonathan-Vasquez committed:
se corrigio el constructor del Jugador con el HashMap

AbrahamOsco committed:
Merge branch 'Entrega2' into abraham

marco-amm committed:
se adaptan los tests de Entrega1 al nuevo modelo despues del hard reset

Jonathan-Vasquez committed:
se corrigio la logica de obtenerPuntaje de PreguntaPuntajeParcial

Jonathan-Vasquez committed:
se agregaron los test para la clase PreguntaClasica

Jonathan-Vasquez committed:
Se refactorizo la familia de Preguntas

Jonathan-Vasquez committed:
Se añadieron los test de las Preguntas PuntajeParcialClasico y Penalizable, y de PreguntaConTodasLasOpcionesCorrectas

Jonathan-Vasquez committed:
Se agregaron los test para la clase PreguntasConOpcionesClasificadas

Jonathan-Vasquez committed:
Se cambio el Evaluable por una Opcion Concreta y se hizo Pregunta groupChoice OrderedChoice y MultipleChoice

marco-amm committed:
se agregan el resto de preguntas y sus test despues del soft reset, falta refactorizar para eliminar codigo duplicado

marco-amm committed:
se agregan losp paquetes donde iran los tests de las  entregas

marco-amm committed:
se vuelven a agregar los test de Entrega1

marco-amm committed:
se agregan los test de multipleChoicePenalizable, orderedChoice y groupChoice a Entrega2Test, faltan los de multiplicadores

Jonathan-Vasquez committed:
Se borro las clases Respuesta y Asignador de puntos y se implementaron las Clases Turno y Ronda

Jonathan-Vasquez committed:
se agrego test02PreguntaDeVerdaderoFalsoPenalizableRecibeLasOpcionesElegidasPorLosJugadoresEntoncesAsignaBienSusPuntos() con Turno y Ronda Funcionando

Jonathan-Vasquez committed:
Se agregaron los test de Entrega0 y Entrega1 utilizando Ronda y Turno

marco-amm committed:
se adaptan los test de clases por que ahora reciben directamente la listaDeListasDeOpciones en lugar de recibirlas dentro de respuesta

marco-amm committed:
Merge branch 'marco' into Entrega2

Jonathan-Vasquez committed:
Se agregaron los test Entrega2 sin multiplicadores

Jonathan-Vasquez committed:
se reorganizaron los test del paquete Entrega2

marco-amm committed:
se agrega Jugador tests y algunos de sus tests unitarios para el metodo activarMultiplicador

Jonathan-Vasquez committed:
Se agregaron los test para entrega2 con usando multiplicidad en PreguntasPenalizables

Jonathan-Vasquez committed:
Merge branch 'Entrega2' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega2

AbrahamOsco committed:
eliminado conflictos

AbrahamOsco committed:
Agregaron Los Test De la Clase Turno

AbrahamOsco committed:
Se refactorizan los paquetes

Commit:
Añadidos Test de Ronda

AbrahamOsco committed:
Merge branch 'abraham' into Entrega2

Jonathan-Vasquez committed:
se refactorizo Pregunta para no repetir Codigo

Jonathan-Vasquez committed:
resolviendo conflictos de rama

Jonathan-Vasquez committed:
se resolvieron conflictos de coolision de commits

Jonathan-Vasquez committed:
se modifico el archivo module-info-java

marco-amm committed:
se pasan las anteriores preguntas y evaluables al nuevo modelo, se adaptan clases y tests de Entregas

marco-amm committed:
Se reemplaza la interfaz Evaluable por la clase abstracta Opcion

marco-amm committed:
se agregan los tests de preguntas

marco-amm committed:
nada

Jonathan-Vasquez committed:
se implemento la clase exclusividad

Jonathan-Vasquez committed:
Se modifco Exclusividad , Ronda,Turno  y se agregaron  algunos  test de Exclusividad

Jonathan-Vasquez committed:
se borro el modificador global y nulo

AbrahamOsco committed:
Se eliminaron las clases ModificadorGlobal y ModificadorNulo

AbrahamOsco committed:
Se agregaron los test de clases multipleChoise Clasica,parcial y ordered choise usando exclusividad

AbrahamOsco committed:
Se agregaron mas pruebas de Ordered Choise y GroupChoise

AbrahamOsco committed:
Agrego archivos necesarios para serializar json.iml y persistencia.iml

AbrahamOsco committed:
Haciendo lectura de archivos agrego la bilbioteca gson , modifico pom,creo una clase de prueba de lectura json

AbrahamOsco committed:
cambios pequeños

marco-amm committed:
casmbiando cola de prioridad por codigo con uso stream

AbrahamOsco committed:
Merge branch 'Entrega2' into abraham

marco-amm committed:
se generaliza y refacotirza codigo repetido de preguntas y cambia el comportamiento de ocpion(ahora devuelve su puntaje) y se agrego opcionIncorrectaPenalizable, con abraham y mateo(aguante el three programminggit add .)

Jonathan-Vasquez committed:
se borro setters de Pregunta de consigna y opciones

marco-amm committed:
se reemplazan los for de los metodos de Pregunta por sus equivalentes usando stream()

marco-amm committed:
quitando getConsigna para no generar conflicto al mergear con Entrega2

marco-amm committed:
problemas de fusion

Jonathan-Vasquez committed:
refactor pregunta getOpcionesCorrectas()

Jonathan-Vasquez committed:
se refactorizo el nombre de variable

Jonathan-Vasquez committed:
se erradicaron todos los for

ElBrujas committed:
agregado getters & setters de la clase Partida

ElBrujas committed:
Agregamos un toString al jugador
Inteamos poner el metodo obtnerPuntajes en Partida

Jonathan-Vasquez committed:
se agrego el metodo obtenerResultadoPartida

Jonathan-Vasquez committed:
se agrego el metodo obtenerResultadoPartida

Jonathan-Vasquez committed:
Merge branch 'Jonathan(Jet)' into Entrega2

Jonathan-Vasquez committed:
arreglo renames

Jonathan-Vasquez committed:
se arreglo el bug del refactor

marco-amm committed:
se agrega un test de partida, faltan mas

Jonathan-Vasquez committed:
se cambio el nombre de la clase Ronda a AsignadorPuntos

Jonathan-Vasquez committed:
se arreglo el metodo obtenerResultadoPartida de Partida

AbrahamOsco committed:
Merge branch 'Entrega2' into abraham

ElBrujas committed:
Creamos y pusimos comportamiento a Juego

Jonathan-Vasquez committed:
refactor en Juego del metodo cargarJugadoresActivos()

edjus commented on src/main/java/edu/fiuba/algo3/modelo/Juego/Juego.java:
```diff
@@ -0,0 +1,103 @@
+	private void cargarJugadoresActivos() {
+		unaPartida.getJugadores().stream().forEach(jugador -> jugadoresActivos.offer(jugador));
+
+		/*for (Jugador unJugador: unaPartida.getJugadores()) {
```
Borrar código comentado

edjus commented on src/main/java/edu/fiuba/algo3/modelo/Juego/Juego.java:
```diff
@@ -0,0 +1,103 @@
+		return unaPartida.obtenerResultadoPartida();
+	}
+	
+	/*public Pantalla obtenerUnaPantalla() {
```
Borrar código comentado

edjus reviewed (approved):
Quedó bastante mejor el modelo, solo vi un par de código comentado.

AbrahamOsco committed:
Se crean las clases para recuperar cada pregunta

AbrahamOsco committed:
Se agregaron test de LecturaDeJson  y se terminaron los metodos para leer los archivos y para devolver una cola de preguntas

Jonathan-Vasquez committed:
prueba fx

Jonathan-Vasquez committed:
se crea directorio para Vistas

Jonathan-Vasquez committed:
Merge branch 'Jonathan(Jet)' into Entrega3

AbrahamOsco committed:
moviendo paquetes

marco-amm committed:
probando fxml, nada q ver :v

marco-amm committed:
scena vista pregunta,pruebaita

AbrahamOsco committed:
Se termino de implementar las clases para leer el archivo Json y Se Agregaron mas Pruebas

AbrahamOsco committed:
Arreglando conflictos y agregando Json a la Entrega 3

AbrahamOsco committed:
Se agrega el repositorio Json

AbrahamOsco committed:
Se hacen un test de JsonLectura y se carga las pregunta en la cola de preguntas del juego

AbrahamOsco committed:
se borro el head del conflicto

ElBrujas committed:
Agregado la clase game launcher

Agregado Las Clases padres de Controlador

marco-amm committed:
pequeño intento de mvc

ElBrujas committed:
Ahora El ControladorSecundario tiene el metodo para cerrar esa escena

ElBrujas committed:
agregadas las vistas

marco-amm committed:
nada

ElBrujas committed:
Merge remote-tracking branch 'origin/Entrega3' into Samuel

Conflicts:
	src/main/java/edu/fiuba/algo3/Vista/App.java

marco-amm committed:
actualizando rama con json de entrega3

ElBrujas committed:
Se agregaron clases a los controladores

testeamos la union de menu con el de crear jugadores

por favor pongan los atributos de los controladores en publico

AbrahamOsco committed:
Arreglando la Rama de Samuel para poder correr la aplicacion se Exporto un modulo mas y se puso un atributo en publico

marco-amm committed:
plan b casi resuelto :v

ElBrujas committed:
agregado a medias el comportamiento del controlador de menucrearjugador

AbrahamOsco committed:
Se agrego un identificador a la clase Pregunta

ElBrujas committed:
Merge remote-tracking branch 'origin/Entrega3' into Samuel2

Jonathan-Vasquez committed:
se agrego el obtenerPantallaParaPregunta en MenuCrearJugadorController

Commit:
Escritas Pre y Pos condiciones en Modelo, exceptuando LecturaJson

Commit:
Añadidos tests positivos en TurnoTests.

Commit:
Revisadas las potenciales excepciones.

ElBrujas committed:
commit de emergencia

ElBrujas committed:
otro commit de emergencia

ElBrujas committed:
ultimo commit de emergencia

marco-amm committed:
plan de emergencia casi completo :v

ElBrujas committed:
commit de emergencia dia 2

ElBrujas committed:
Implementamos la navegacion entre ventanas

La visibilidad de los modificadores funcan piola

ElBrujas committed:
Unimos los contoladores de las pantallas de juego en uno solo

AbrahamOsco committed:
Se desordena el arrayList de pregunta que se leen del archivo

AbrahamOsco committed:
arreglo conflictos

AbrahamOsco committed:
arrgland mas conflictos

AbrahamOsco committed:
Se agrego ahora que la lista devuelve preguntas aleatorias del archivo

ElBrujas committed:
testeando la wea, pero no asigna puntos con las ordered chois

AbrahamOsco committed:
se arreglan conflictos

AbrahamOsco committed:
se resuelve ultimos conflcitos

Jonathan-Vasquez committed:
se refactorizo GameLauncher

Jonathan-Vasquez committed:
arreglando las porquerias de Samuel

Jonathan-Vasquez committed:
se cambiaron nombres a algunas variables de la clase InicializadorOpciones

AbrahamOsco committed:
guardo cambios

AbrahamOsco committed:
guarda cambios

ElBrujas committed:
Ya funcionoan los modificadores en pantalla

AbrahamOsco committed:
guardo cambios

ElBrujas committed:
Merge remote-tracking branch 'origin/Entrega3' into Samuel2

Conflicts:
	src/main/java/edu/fiuba/algo3/control/InicializadorOpciones.java
	src/main/java/edu/fiuba/algo3/modelo/Juego/Juego.java

AbrahamOsco committed:
aarregland conflic

Jonathan-Vasquez committed:
se arreglo la lectura del json de las preguntas orderedChoice

AbrahamOsco committed:
Se corrige el orderedChoice

AbrahamOsco committed:
Se arreglan conflictos

ElBrujas committed:
Fixeamos los comportamientos de los botones de las vista de

OrderedChoice y GroupChoice

AbrahamOsco committed:
Merge branch 'Samuel2' into Abraham

ElBrujas committed:
arreglado los problemas con los modificadores

AbrahamOsco committed:
Se guardan pequeños

ElBrujas committed:
Ahora la pantalla detecta empates, sin importar la longitud del nombre

ElBrujas committed:
commit de emergencia, programa funcionando, faltan las trancisiones.

ElBrujas committed:
Pantalla de transicion a medio hacer

AbrahamOsco committed:
Se agregan mas test en Json y se tiene un Json especial para los test

ElBrujas committed:
Ya funciona la pantalla de cambio de Jugador, bueno, a las chapas

ElBrujas committed:
Juego funcionando, ya es GOLD

ElBrujas committed:
Revert "se arreglo la lectura del json de las preguntas orderedChoice"

This reverts commit a4a293d668ca21565b733fb790d2e49358d260ae.

AbrahamOsco committed:
Merge branch 'Entrega3' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega3

Jonathan-Vasquez committed:
se corrigieron algunos nombres en el modelo

ElBrujas committed:
Juego terminado subido a la rama3

Jonathan-Vasquez committed:
se corrigen cosas para mergear

Jonathan-Vasquez committed:
Merge branch 'Entrega3' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega3

AbrahamOsco committed:
se eliminan  algunas clases que no se usan

AbrahamOsco committed:
Merge branch 'Entrega3' of https://github.com/AbrahamOsco/tp2Algo3 into Entrega3

AbrahamOsco committed:
Se termino de eliminar archivos que no se usan

AbrahamOsco committed:
Se termino la limpieza de clases que no se usan
